packages/datastore/datastore/src/replication/mutations/QueueUpdateProcessor.ts

```diff
@@ -0,0 +1,138 @@
+import { ModelMap } from "./MutationsQueue";
+import { MutationRequest } from "./MutationRequest";
+import { LocalStorage, CRUDEvents } from "../../storage";
+import { Model } from "../../Model";
+import traverse from "traverse";
```

Comment:
why traverse is needed?

Reply:
It allows us to swap the client IDs in place anywhere in the queue items array